controllers/util/mw_util.go

```diff
@@ -200,16 +201,76 @@ func (mwu *MWUtil) generateVRGManifest(
 	})
 }
 
-func (mwu *MWUtil) CreateOrUpdateVRGRolesManifestWork(namespace string) error {
-	manifestWork, err := mwu.generateVRGRolesManifestWork(namespace)
+func (mwu MWUtil) ClusterRolesList(clusterNames *sets.String) error {
+	manifestworks := &ocmworkv1.ManifestWorkList{}
+	if err := mwu.Client.List(mwu.Ctx, manifestworks); err != nil {
+		return fmt.Errorf("manifestworks list: %w", err)
+	}
+
+	for i := range manifestworks.Items {
+		manifestwork := &manifestworks.Items[i]
+		if manifestwork.ObjectMeta.Name == ClusterRolesManifestWorkName {
+			*clusterNames = clusterNames.Insert(manifestwork.ObjectMeta.Namespace)
+		}
+	}
+
+	return nil
+}
+
+var clusterRolesMutex sync.Mutex
+
+func (mwu *MWUtil) ClusterRolesCreate(drpolicy *rmn.DRPolicy) error {
+	clusterRolesMutex.Lock()
```

Comment:
why do you need a lock?  ClusterRoles are created from the reconciler and I believe that thread safe

Reply:
It is my understanding that a hub can have multiple DRPolicies each with at most one reconciler running at a time.  The lock serializes concurrent creates, deletes, or creates and deletes.

Reply:
Right, but I am not sure what the lock is protecting.  I have to think about it.  Ignore my comment for now.

Reply:
Suppose drpolicyA specifies cluster1 and is being deleted, while drpolicyB also specifying cluster1 is being created.  drpolicyA's reconciler reads all drpolicies and finds no other drpolicies include cluster1 because drpolicyB has not yet been created, so it determines cluster1's roles can be deleted.  But before it does so, drpolicyB is created and its reconciler determines cluster1's roles need not be created because it already exists, then drpolicyA's reconciler deletes the roles that drpolicyB expects to exist.   The mutex makes the read-modify-write operation atomic.